src/common/BaseLightAccount.sol

```diff
@@ -0,0 +1,139 @@
+        _onlyOwnerOrEntryPoint();
+        _call(dest, value, func);
+    }
+
+    /// @notice Execute a sequence of transactions.
+    /// @param dest An array of the targets for each transaction in the sequence.
+    /// @param func An array of calldata for each transaction in the sequence. Must be the same length as `dest`, with
+    /// corresponding elements representing the parameters for each transaction.
+    function executeBatch(address[] calldata dest, bytes[] calldata func) external virtual {
+        _onlyOwnerOrEntryPoint();
+        if (dest.length != func.length) {
+            revert ArrayLengthMismatch();
+        }
+        uint256 length = dest.length;
+        for (uint256 i = 0; i < length; ++i) {
+            _call(dest[i], 0, func[i]);
+        }
+    }
+
+    /// @notice Execute a sequence of transactions.
+    /// @param dest An array of the targets for each transaction in the sequence.
+    /// @param value An array of value for each transaction in the sequence.
+    /// @param func An array of calldata for each transaction in the sequence. Must be the same length as `dest`, with
+    /// corresponding elements representing the parameters for each transaction.
+    function executeBatch(address[] calldata dest, uint256[] calldata value, bytes[] calldata func) external virtual {
+        _onlyOwnerOrEntryPoint();
+        if (dest.length != func.length || dest.length != value.length) {
+            revert ArrayLengthMismatch();
+        }
+        uint256 length = dest.length;
+        for (uint256 i = 0; i < length; ++i) {
+            _call(dest[i], value[i], func[i]);
+        }
+    }
+
+    /// @notice Deposit more funds for this account in the entry point.
+    function addDeposit() public payable {
+        entryPoint().depositTo{value: msg.value}(address(this));
+    }
+
+    /// @notice Withdraw value from the account's deposit.
+    /// @param withdrawAddress Target to send to.
+    /// @param amount Amount to withdraw.
+    function withdrawDepositTo(address payable withdrawAddress, uint256 amount) public onlyOwner {
+        entryPoint().withdrawTo(withdrawAddress, amount);
+    }
+
+    /// @inheritdoc BaseAccount
+    function entryPoint() public view virtual override returns (IEntryPoint) {
+        return _ENTRY_POINT;
+    }
+
+    /// @notice Check current account deposit in the entry point.
+    /// @return The current account deposit.
+    function getDeposit() public view returns (uint256) {
+        return entryPoint().balanceOf(address(this));
+    }
+
+    /// @inheritdoc IERC1271
+    /// @dev Must override to support ERC-1271 signature validation.
+    function isValidSignature(bytes32 hash, bytes memory signature) public view virtual override returns (bytes4);
+
+    /// @dev Must override to support user op validation.
+    function _validateSignature(PackedUserOperation calldata userOp, bytes32 userOpHash)
+        internal
+        virtual
+        override
+        returns (uint256);
```

Comment:
Do we get this unimplemented abstract function from `BaseAccount` already?

Reply:
Yup, nice catch